test_communication/test/subscriber_py.py

```diff
@@ -42,17 +42,20 @@ def listener_cb(msg, received_messages, expected_msgs):
         raise RuntimeError('received unexpected message %r' % msg)
 
 
-def listener(message_name, rmw_implementation, number_of_cycles):
+def listener(message_name, number_of_cycles):
+    from message_fixtures import get_test_msg
     import rclpy
-    from rclpy.qos import qos_profile_default
     from rclpy.impl.rmw_implementation_tools import select_rmw_implementation
-    from message_fixtures import get_test_msg
+    from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+    from rclpy.qos import qos_profile_default
 
     message_pkg = 'test_communication'
     module = importlib.import_module(message_pkg + '.msg')
     msg_mod = getattr(module, message_name)
 
-    select_rmw_implementation(rmw_implementation)
+    rmw_implementations = get_rmw_implementations()
+    assert(os.environ['RCLPY_IMPLEMENTATION'] in rmw_implementations)
```

Comment:
Good point this way we will either raise a `KeyError` exception if the env variable doesn't exist or a `InvalidRCLPYImplementation` exception if the content is not valid.
Fixed in https://github.com/ros2/system_tests/pull/152/commits/55d6d75f2684f78d3ae17b508ac1c4bbeba893c8

rcl variables renamed in https://github.com/ros2/system_tests/pull/152/commits/fd14f0149fa71f0890539f259ba87d7381ac4c2c